src/rule_config.jl

```diff
@@ -14,3 +14,38 @@ function ChainRulesCore.rrule_via_ad(config::ADviaRuleConfig, f, args...; kws...
     ret === nothing && throw(MethodError(rrule, (f, args...)))
     return ret
 end
+
+# For testing this config uses finite differences to evaluate the frule and rrule
+struct ADviaFDConfig <: RuleConfig{Union{HasReverseMode, HasForwardsMode}}
+    fdm
+end
+
+function ChainRulesCore.frule_via_ad(config::ADviaFDConfig, ȧrgs, f, args...; kws...)
+
+    call_on_copy(f, xs...) = deepcopy(f)(deepcopy(xs)...; deepcopy(kws)...)
+
+    primals = (f, args...)
+    is_ignored = isa.(ȧrgs, NoTangent)
+
+    Ω = call_on_copy(f, args...)
+    ΔΩ = _make_jvp_call(config.fdm, call_on_copy, Ω, primals, ȧrgs, is_ignored)
+
+    return Ω, ΔΩ
+end
+
+function ChainRulesCore.rrule_via_ad(config::ADviaFDConfig, f, args...; kws...)
+
+    call(f, xs...) = f(xs...; kws...)
+
+    # this block is here just to work out which tangents should be ignored
+    primals_and_tangents = auto_primal_and_tangent.((f, args...))
+    primals = primal.(primals_and_tangents)
+    accum_cotangents = tangent.(primals_and_tangents)
+    is_ignored = isa.(accum_cotangents, NoTangent)
```

Comment:
Shall we avoid creating a variable that is never used again?
The following fuses
```suggestion
    is_ignored = isa.(tangent.(primals_and_tangents), NoTangent)
```